server.js

```diff
@@ -16,12 +17,83 @@ const app = express();
 app.use(cors());
 app.use(express.json());
 
+// Thought schema
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    minlength: 5,
+    maxlength: 400,
+    required: true
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+// Model for ThoughtSchema
+const Thought = mongoose.model('Thought', ThoughtSchema)
+
+
 // Start defining your routes here
+// GET: List of endpoints
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(listEndpoints(app))
 });
 
+// GET: List of 20 most recent thoughts, in descending order.
+app.get("/thoughts", async (req, res) => {
+  const thoughts = await Thought.find().sort({ createdAt: 'desc' }).limit(20).exec();
```

Comment:
Is desc working for you? I found an explanation that  In JavaScript, desc is not a valid keyword for sorting in MongoDB. Instead, I used -1 to sort in descending order based on the createdAt field. But looks like it works for you!